osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            Height = 30;
+            RelativeSizeAxes = Axes.X;
+
+            Children = new Drawable[]
+            {
+                hoverBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider!.Background3,
+                    Alpha = 0f,
+                },
+                selectBox = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider!.Background4,
+                    Alpha = 0f,
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Padding = new MarginPadding { Left = 18, Right = 5 },
+                    Child = new GridContainer
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        ColumnDimensions = new[]
+                        {
+                            new Dimension(GridSizeMode.AutoSize),
+                            new Dimension(),
+                            new Dimension(GridSizeMode.AutoSize),
+                            new Dimension(GridSizeMode.AutoSize),
+                        },
+                        Content = new[]
+                        {
+                            new Drawable[]
+                            {
+                                createAvatar(),
+                                text = new ControlItemText(channel)
+                                {
+                                    Anchor = Anchor.CentreLeft,
+                                    Origin = Anchor.CentreLeft,
+                                },
+                                mention = new ControlItemMention
+                                {
+                                    Anchor = Anchor.CentreLeft,
+                                    Origin = Anchor.CentreLeft,
+                                    Margin = new MarginPadding { Right = 3 },
+                                },
+                                close = new ControlItemClose
+                                {
+                                    Anchor = Anchor.CentreLeft,
+                                    Origin = Anchor.CentreLeft,
+                                    Margin = new MarginPadding { Right = 3 },
+                                }
+                            }
+                        },
+                    },
+                },
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            selectedChannel?.BindValueChanged(change =>
+            {
+                if (change.NewValue == channel)
+                    selectBox?.Show();
+                else
+                    selectBox?.Hide();
+            }, true);
+
+            Action = () => OnRequestSelect?.Invoke(channel);
+            close!.Action = () => OnRequestLeave?.Invoke(channel);
```

Comment:
Going to be interested in other opinions on this because I have had the same problem - this sort of assignment with null-forgiveness is what initialising drawables in BDL with nullability enabled inevitably leads to, and to me this reads really badly.

Reply:
In this specific case, the actiosn could be assigned on object initializer rather than here, i.e.:
```diff
diff --git a/osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs b/osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs
index e475ac7821..4d989ebdfd 100644
--- a/osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs
+++ b/osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs
@@ -114,12 +114,15 @@ private void load()
                                     Anchor = Anchor.CentreLeft,
                                     Origin = Anchor.CentreLeft,
                                     Margin = new MarginPadding { Right = 3 },
+                                    Action = () => OnRequestLeave?.Invoke(channel),
                                 }
                             }
                         },
                     },
                 },
             };
+
+            Action = () => OnRequestSelect?.Invoke(channel);
         }

         protected override void LoadComplete()
@@ -133,9 +136,6 @@ protected override void LoadComplete()
                 else
                     selectBox?.Hide();
             }, true);
-
-            Action = () => OnRequestSelect?.Invoke(channel);
-            close!.Action = () => OnRequestLeave?.Invoke(channel);
         }

         protected override bool OnHover(HoverEvent e)
```

But in any other case, I think it's safe to say those BDL fields will be non-null in any other place in the component itself, so they can be declared as non-nullabe with `= null!` appended at the end.

Reply:
You'd say that [but I've been told differently otherwise in the past](https://github.com/ppy/osu/pull/16264#discussion_r775506750). Also see [this thread](https://github.com/ppy/osu/pull/17014#discussion_r816237281).

Reply:
Interesting opinions, I can rarely think of any local `Drawable` component that has logic executed before BDL myself, so thought using `= null!` makes sense. I guess `AsNonNull()` is the way then...? I'm not sure.

Reply:
We've started to do the same in realm as well. Difference here is that realm guarantees to populate in the ctor, while here we lose null protection for things like properties which may access a field before BDL has been run.

Reply:
I'd still like to see these particular assignments of `Action` to be moved to BDL as proposed above.

Reply:
We've come to the general conclusion of using `=null!` suppression for BDL usage, so you're welcome to apply that here if you'd like to.